exist-core/src/main/java/org/exist/dom/persistent/VirtualNodeSet.java

```diff
@@ -113,6 +113,32 @@ public boolean contains(final NodeProxy p) {
         return false;
     }
 
+    @Override
+    public boolean containsReference(final Item item) {
+        if (!(item instanceof NodeProxy)) {
+            return false;
+        }
+
+        final NodeProxy firstParent = getFirstParent((NodeProxy) item, null, axis == Constants.SELF_AXIS, 0);
+        if (firstParent == item) {
+            return true;
+        }
+        return false;
```

Comment:
Codacy error could be fixed by writing:
`return firstParent == item`

Reply:
Okay fixed.